apis/configwatch.go

```diff
@@ -0,0 +1,71 @@
+// Copyright Contributors to the L3AF Project.
+// SPDX-License-Identifier: Apache-2.0
+//
+// +build !configs
+//
+
+package apis
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/l3af-project/l3afd/config"
+	"github.com/l3af-project/l3afd/kf"
+	"github.com/l3af-project/l3afd/routes"
+	"github.com/rs/zerolog/log"
+)
+
+type Server struct {
+	KFRTConfigs *kf.NFConfigs
+	HostName    string
+}
+
+func StartConfigWatcher(ctx context.Context, hostname, daemonName string, conf *config.Config, kfrtconfg *kf.NFConfigs) error {
+	log.Info().Msgf("%s config server setup started on host %s", daemonName, hostname)
+
+	s := &Server{
+		KFRTConfigs: kfrtconfg,
+		HostName:    hostname,
+	}
+
+	term := make(chan os.Signal)
+	signal.Notify(term, syscall.SIGTERM, syscall.SIGINT)
```

Comment:
this line looks Linux specific, not cross-plat. See PR #25 for the cross-plat API created in that PR.